### PR TITLE
chore: fix recognition of internal links + post rate limiting

### PR DESCRIPTION
**Pull request by taggrx**

### CODE REVIEW

**Comment by chmllr** [comment]
LGTM

File: src/backend/env/post.rs [hunk 1]
```diff
@@ -463,31 +463,38 @@ impl Post {
...
-            CONFIG.max_posts_per_hour
-        } else {
-            CONFIG.max_comments_per_hour
-        } as usize;
+        if user
+            .posts(state, 0)
+            .filter(|post| {
+                parent.is_none()
+                    && post.parent.is_none()
+                    && post.timestamp() > timestamp.saturating_sub(DAY)
```
Comment (leo-ship-it) [inline src/backend/env/post.rs:471]: Here I think you are checking for posts that are 1 DAY or older.

File: src/backend/env/post.rs [hunk 2]
```diff
@@ -463,31 +463,38 @@ impl Post {
...
+            ));
+        }
 
         if user
             .posts(state, 0)
             .filter(|post| {
-                !(parent.is_none() ^ post.parent.is_none())
+                parent.is_some()
+                    && post.parent.is_some()
                     && post.timestamp() > timestamp.saturating_sub(HOUR)
```
Comment (leo-ship-it) [inline src/backend/env/post.rs:487]: Here same, aren't you counting posts that are 1 hour or older?

File: src/backend/env/post.rs (hunk 1 above)
Comment (taggrx) [reply src/backend/env/post.rs:471]: No, I think it's 1 DAY or newer: posts, which have a timestamp above `today - one day`, right?

File: src/backend/env/post.rs (hunk 2 above)
Comment (taggrx) [reply src/backend/env/post.rs:487]: No, we're rate limiting. For that I need to know how many posts/comments were produced in the last hour/day, right?

File: src/backend/env/post.rs (hunk 1 above)
Comment (leo-ship-it) [reply src/backend/env/post.rs:471]: You're correct, my bad.

**Pull request merged**